ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ExecuteQueryResponseHandler.java

```diff
@@ -58,31 +62,50 @@
   }
 
   @Override
-  protected void handleMetadata(final QueryResponseMetadata queryResponseMetadata) {
-    cf.queryID().complete(queryResponseMetadata.queryId);
-    columnNames = queryResponseMetadata.columnNames;
-    columnTypes = RowUtil.columnTypesFromStrings(queryResponseMetadata.columnTypes);
+  protected void handleMetadata(final StreamedRow queryResponseMetadata) {
+    final String queryId = queryResponseMetadata.getHeader().get().getQueryId().toString();
+
+    cf.queryID().complete(queryId);
+
+    final List<Column> columns = queryResponseMetadata
```

Comment:
Share this logic with the class below?  It's pretty simple, but still might make sense to push into the parent class.